pilot/pkg/security/model/authentication_test.go

```diff
@@ -59,68 +50,64 @@ func TestConstructSdsSecretConfig(t *testing.T) {
 							ApiType: core.ApiConfigSource_GRPC,
 							GrpcServices: []*core.GrpcService{
 								{
-									TargetSpecifier: &core.GrpcService_EnvoyGrpc_{
-										EnvoyGrpc: &core.GrpcService_EnvoyGrpc{ClusterName: SDSClusterName},
-									},
-								},
-							},
-							RefreshDelay: nil,
-						},
-					},
-				},
-			},
-		},
-		{
-			serviceAccount: "spiffe://cluster.local/ns/bar/sa/foo",
-			sdsUdsPath:     "/tmp/sdsuds.sock",
-			expected: &auth.SdsSecretConfig{
-				Name: "spiffe://cluster.local/ns/bar/sa/foo",
-				SdsConfig: &core.ConfigSource{
-					InitialFetchTimeout: features.InitialFetchTimeout,
-					ConfigSourceSpecifier: &core.ConfigSource_ApiConfigSource{
-						ApiConfigSource: &core.ApiConfigSource{
-							ApiType: core.ApiConfigSource_GRPC,
-							GrpcServices: []*core.GrpcService{
-								{
-									TargetSpecifier: &core.GrpcService_EnvoyGrpc_{
-										EnvoyGrpc: &core.GrpcService_EnvoyGrpc{ClusterName: SDSClusterName},
+									TargetSpecifier: &core.GrpcService_GoogleGrpc_{
+										GoogleGrpc: &core.GrpcService_GoogleGrpc{
+											TargetUri:  "/tmp/sdsuds.sock",
+											StatPrefix: SDSStatPrefix,
+										},
 									},
 								},
 							},
-							RefreshDelay: nil,
 						},
 					},
 				},
 			},
 		},
 		{
+			name:           "CustomUds without service account",
 			serviceAccount: "",
 			sdsUdsPath:     "/tmp/sdsuds.sock",
 			expected:       nil,
 		},
 		{
-			serviceAccount: "",
-			sdsUdsPath:     "spiffe://cluster.local/ns/bar/sa/foo",
+			name:           "CustomUds without sdsUdsPath",
+			serviceAccount: "spiffe://cluster.local/ns/bar/sa/foo",
+			sdsUdsPath:     "",
 			expected:       nil,
 		},
 	}
 
 	for _, c := range cases {
-		if got := ConstructSdsSecretConfig(c.serviceAccount, c.sdsUdsPath); !reflect.DeepEqual(got, c.expected) {
-			t.Errorf("ConstructSdsSecretConfig: got(%#v) != want(%#v)\n", got, c.expected)
-			fmt.Println(got)
-			fmt.Println(c.expected)
-		}
+		t.Run(c.name, func(t *testing.T) {
+			if got := ConstructSdsSecretConfigWithCustomUds(c.serviceAccount, c.sdsUdsPath); !cmp.Equal(got, c.expected, protocmp.Transform()) {
+				t.Errorf("ConstructSdsSecretConfigWithCustomUds: got(%#v), want(%#v)\n", got, c.expected)
+			}
+		})
 	}
 }
 
-func TestConstructSdsSecretConfigWithCustomUds(t *testing.T) {
+func TestConstructSdsSecretConfig(t *testing.T) {
+	gRPCConfig := &core.GrpcService_GoogleGrpc{
+		TargetUri:  "/tmp/sdsuds.sock",
+		StatPrefix: SDSStatPrefix,
+		ChannelCredentials: &core.GrpcService_GoogleGrpc_ChannelCredentials{
+			CredentialSpecifier: &core.GrpcService_GoogleGrpc_ChannelCredentials_LocalCredentials{
+				LocalCredentials: &core.GrpcService_GoogleGrpc_GoogleLocalCredentials{},
+			},
+		},
+	}
+
+	gRPCConfig.CredentialsFactoryName = FileBasedMetadataPlugName
+	gRPCConfig.CallCredentials = ConstructgRPCCallCredentials(K8sSATrustworthyJwtFileName, K8sSAJwtTokenHeaderKey)
```

Comment:
is `gRPCConfig` ever used in the function?

Reply:
This part is not my code, see origin file line 31.
I just reorder the tests order to make them align to authentication.go

Reply:
can you remove it as you're touching this file?